codebase/src/java/disco/org/openlvc/disco/configuration/RprConfiguration.java

```diff
@@ -445,30 +468,88 @@ private void loadDefaultModules()
-		    "hla/rpr2/RPR-DER_v2.0.xml",
-		    "hla/rpr2/RPR-Enumerations_v2.0.xml",
-		    "hla/rpr2/RPR-Physical_v2.0.xml",
-		    "hla/rpr2/RPR-SIMAN_v2.0.xml",
-		    "hla/rpr2/RPR-Warfare_v2.0.xml"
+			"hla/rpr2/HLAstandardMIM.xml",
+			"hla/rpr2/RPR-Foundation_v2.0.xml",
+			"hla/rpr2/RPR-Base_v2.0.xml",
+			"hla/rpr2/RPR-Communication_v2.0.xml",
+			"hla/rpr2/RPR-DER_v2.0.xml",
+			"hla/rpr2/RPR-Enumerations_v2.0.xml",
+			"hla/rpr2/RPR-Physical_v2.0.xml",
+			"hla/rpr2/RPR-SIMAN_v2.0.xml",
+			"hla/rpr2/RPR-Warfare_v2.0.xml"
 		};
-
-		// Make sure we can find each default
-		ClassLoader loader = getClass().getClassLoader();
-		for( String module : defaultModules )
+		
+		if( getRtiProvider() == RtiProvider.Mak )
 		{
-			URL url = loader.getResource( module );
-			if( url == null )
-				throw new DiscoException( "Could not find FOM module: "+url );
-			else
-				fomModules.add( url );
+			createMakFomFiles( defaultModules );
+		}
+		else
+		{
+			// Make sure we can find each default
+			ClassLoader loader = getClass().getClassLoader();
+			for( String module : defaultModules )
+			{
+				URL url = loader.getResource( module );
+				if( url == null  )
+					throw new DiscoException( "Could not find FOM module: "+url );
+				else
+					fomModules.add( url );
+			}
+		}
+	}
+	
+	/**
+	 * Mak refuses to load anything from a jar file, as such need to copy any
+	 * module files in the jar outside into a folder where Mak can load them.
+	 * This method does that, creating folders for the rpr2 and dcss modules
+	 * in the install directory, copying in the given module files, and adding
+	 * them to the list of module URLs
+	 */
+	private void createMakFomFiles( String[] modules )
+	{
+		try
+		{
+			// create module directories if not already there
+			File rprDir = new File( "hla/rpr2" );
+			Files.createDirectories( rprDir.toPath() );
+			File dcssDir = new File( "hla/dcss" );
+			Files.createDirectories( dcssDir.toPath() );
+		}
+		catch( IOException e )
+		{
+			throw new DiscoException( "Error creating module directories", e );
+		}
+		
+		try
+		{
+			ClassLoader loader = getClass().getClassLoader();
+			for( String module : modules )
+			{
+				// Mak doesn't like this module being included
+				if( module.contains( "HLAstandardMIM") )
+					continue;
+				
+				// load resource as normal, but then stream to a regular file so Mak can use it
+				URL url = loader.getResource( module );
+				File moduleFile = new File( module );
+				if( !moduleFile.exists() )
+				{
+					URLConnection con = url.openConnection();
+					InputStream stream = con.getInputStream();
+					Files.copy( stream, moduleFile.toPath(), StandardCopyOption.REPLACE_EXISTING );
+				}
+				
+				// add the url of the new (non-jar) file to the modules list
+				URL newUrl = moduleFile.toURI().toURL();
+				fomModules.add( newUrl );
+			}
+		}
+		catch( IOException e )
+		{
+			throw new DiscoException( "Error loading modules for Mak", e );
```

Comment:
Same as above, can we move this to a generic method (in `FileUtils.java` probably?) that will extract a set of relative path names from a generic jar and copy them to an arbitrary location?